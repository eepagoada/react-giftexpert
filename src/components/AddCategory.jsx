import { useState } from "react"
import PropTypes from 'proptypes'


export const AddCategory = ({onAddCategory}) => {

    const [inputValue, setinputValue] = useState('')

    const onInputChange = ({ target }) => {        
        setinputValue(target.value)
    }

    const onSubmit = (event) =>{

        const value = inputValue.trim()

        event.preventDefault()
        if (value.length <= 1) return

        onAddCategory(value)
        setinputValue("")
    }

    return (
        <form onSubmit={ onSubmit} aria-label="form">
            <input 
                type={"text"}
                placeholder="Buscar gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    onAddCategory : PropTypes.func.isRequired
}
