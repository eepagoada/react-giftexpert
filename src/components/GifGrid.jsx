import { useFetchGifs } from "../hooks/useFetchGifs"
import { GifItem } from "./GifItem"
import PropTypes from 'proptypes'

export const GifGrid = ({category}) => {

    const {isLoading, Image} = useFetchGifs( category )
    
    return (
        <>
            <h3 >{ category }</h3>
            {
                isLoading && <h2>Cargando...</h2>
            }
            <div className="card-grid">
                { 
                    Image.map ( (image) => 
                        <GifItem key={image.id} {...image}/>
                    )
                }
            </div>         
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}
