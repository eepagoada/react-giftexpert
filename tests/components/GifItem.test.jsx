import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components"

describe('pruebas en GifItem', () => { 

    const title = "Enrrin Pagoada"
    const url = "http://www.imagen.com/enrrin.jpg"

    test ('Es igual al snapshot', () => {
        
        const {container} = render( 
            <GifItem title={title} url={url}/>
        )

        expect(container).toMatchSnapshot()
    })

    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => { 
        render(
            <GifItem title={title} url = {url} />
        )

        expect(screen.getByRole('img').src).toBe(url)
        expect(screen.getByRole('img').alt).toBe(title)

    })

    test('Debe de mostrar el titulo en el componente', () => { 
    
        render(
            <GifItem title={title} url = {url} />
        )

        expect(screen.getByText( title )).toBeTruthy()

    })

 })