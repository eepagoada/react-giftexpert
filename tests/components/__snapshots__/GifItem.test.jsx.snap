// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifItem Es igual al snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Enrrin Pagoada"
      src="http://www.imagen.com/enrrin.jpg"
    />
    <p>
      Enrrin Pagoada
    </p>
  </div>
</div>
`;
